zcash_client_sqlite/src/lib.rs

```diff
@@ -208,18 +221,27 @@ mod tests {
         // Create a fake Note for the change
         ctx.outputs.push({
             let change_addr = extfvk.default_address().unwrap().1;
+            let rseed =
+                if consensus::MainNetwork.is_nu_active(NetworkUpgrade::Canopy, height as u32) {
+                    let mut buffer = [0u8; 32];
+                    &rng.fill_bytes(&mut buffer);
+                    Rseed::AfterZip212(buffer)
+                } else {
+                    Rseed::BeforeZip212(Fs::random(&mut rng))
+                };
```

Comment:
This code is repeated 4 times; maybe it can be factored into a utility function?

Reply:
Note that it's also used in `impl SaplingOutput`, not just in test code.

Reply:
refactored into `generate_random_rseed()` method in `zcash_primitives::util`